### PR TITLE
Fix/siriusadapterasan Address sanitizer issues with SiriusAdapter fixed

### LINKED ISSUE #2791: Test and AddressSanitizer error in SiriusAdapter
```
Progress of 'loading spectra list':
-- done [took 0.02 s (CPU), 0.03 s (Wall)] -- 
=================================================================
==24149==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x6020000b5020 at pc 0x40d68a bp 0x7ffe75ad7c00 sp 0x7ffe75ad7bf0
READ of size 8 at 0x6020000b5020 thread T0
    #0 0x40d689 in OpenMS::DPosition<1u, double>::operator[](unsigned long) const /home/jenkins/ws/openms_nightly_supplement/fd57e9df/source/src/openms/include/OpenMS/DATASTRUCTURES/DPosition.h:140
    #1 0x408162 in OpenMS::Peak1D::getMZ() const /home/jenkins/ws/openms_nightly_supplement/fd57e9df/source/src/openms/include/OpenMS/KERNEL/Peak1D.h:116
    #2 0x7f425fd5ce5a in OpenMS::SiriusMSFile::store(OpenMS::MSExperiment const&, OpenMS::String&) /home/jenkins/ws/openms_nightly_supplement/fd57e9df/source/src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp:237
    #3 0x40bed0 in TOPPSiriusAdapter::main_(int, char const**) /home/jenkins/ws/openms_nightly_supplement/fd57e9df/source/src/utils/SiriusAdapter.cpp:220
    #4 0x7f4260277b50 in OpenMS::TOPPBase::main(int, char const**) /home/jenkins/ws/openms_nightly_supplement/fd57e9df/source/src/openms/source/APPLICATIONS/TOPPBase.cpp:491
    #5 0x40764a in main /home/jenkins/ws/openms_nightly_supplement/fd57e9df/source/src/utils/SiriusAdapter.cpp:314
    #6 0x7f425d57d82f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)
    #7 0x4074a8 in _start (/home/jenkins/ws/openms_nightly_supplement/fd57e9df/build/bin/SiriusAdapter+0x4074a8)

0x6020000b5020 is located 0 bytes to the right of 16-byte region [0x6020000b5010,0x6020000b5020)
allocated by thread T0 here:
    #0 0x7f4261bc3eaf in operator new(unsigned long) (/usr/lib/x86_64-linux-gnu/libasan.so.1+0x57eaf)
    #1 0x7f425f60b1c8 in __gnu_cxx::new_allocator<OpenMS::Peak1D>::allocate(unsigned long, void const*) /usr/include/c++/4.9/ext/new_allocator.h:104
    #2 0x7f425f6080bc in std::allocator_traits<std::allocator<OpenMS::Peak1D> >::allocate(std::allocator<OpenMS::Peak1D>&, unsigned long) /usr/include/c++/4.9/bits/alloc_traits.h:488
    #3 0x7f425f605c7b in std::_Vector_base<OpenMS::Peak1D, std::allocator<OpenMS::Peak1D> >::_M_allocate(unsigned long) /usr/include/c++/4.9/bits/stl_vector.h:170
    #4 0x7f425f603929 in void std::vector<OpenMS::Peak1D, std::allocator<OpenMS::Peak1D> >::_M_emplace_back_aux<OpenMS::Peak1D const&>(OpenMS::Peak1D const&) /usr/include/c++/4.9/bits/vector.tcc:412
    #5 0x7f425f6013f9 in std::vector<OpenMS::Peak1D, std::allocator<OpenMS::Peak1D> >::push_back(OpenMS::Peak1D const&) /usr/include/c++/4.9/bits/stl_vector.h:923
    #6 0x7f425fd5c980 in OpenMS::SiriusMSFile::store(OpenMS::MSExperiment const&, OpenMS::String&) /home/jenkins/ws/openms_nightly_supplement/fd57e9df/source/src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp:190
    #7 0x40bed0 in TOPPSiriusAdapter::main_(int, char const**) /home/jenkins/ws/openms_nightly_supplement/fd57e9df/source/src/utils/SiriusAdapter.cpp:220
    #8 0x7f4260277b50 in OpenMS::TOPPBase::main(int, char const**) /home/jenkins/ws/openms_nightly_supplement/fd57e9df/source/src/openms/source/APPLICATIONS/TOPPBase.cpp:491
    #9 0x40764a in main /home/jenkins/ws/openms_nightly_supplement/fd57e9df/source/src/utils/SiriusAdapter.cpp:314
    #10 0x7f425d57d82f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)

SUMMARY: AddressSanitizer: heap-buffer-overflow /home/jenkins/ws/openms_nightly_supplement/fd57e9df/source/src/openms/include/OpenMS/DATASTRUCTURES/DPosition.h:140 OpenMS::DPosition<1u, double>::operator[](unsigned long) const
```

### PR DESCRIPTION
**Pull request by lukaszimmermann**
Fixes #2791 

Problem was in SiriusMSConverter, which assumed the presence of three isotopes. This is now explicitly checked for. @oliveralka  can you check whether the changes make sense?

### CODE REVIEW

**Comment by oliveralka** [comment]
Changes make sense - what about the tests?

File: src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp [hunk 1]
```diff
@@ -225,31 +222,53 @@ namespace OpenMS
         os << ">charge " << int_charge << "\n\n";
 
         // Use precursor m/z & int and no ms1 spectra is available else use values from ms1 spectrum
-        if (isotopes.empty() == false) //if ms1 spectrum was present
+        Size no_isotopes = isotopes.size();
+
+        if ( no_isotopes > 0) //if ms1 spectrum was present
         {
-          os << ">ms1" << "\n";
+          os << ">ms1" << endl;
```
Comment (oliveralka) [inline src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp]: endl might be better than \n - in that case.

File: src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp [hunk 2]
```diff
@@ -225,31 +222,53 @@ namespace OpenMS
...
             }
-          os << "\n";
+          }
+          os << endl;
         }
         else
         {
           if (precursor_int != 0) // if no ms1 spectrum was present but precursor intensity is known
           {
             os << ">ms1" << "\n"
```
Comment (oliveralka) [inline src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp]: endl ?

File: src/utils/SiriusAdapter.cpp [hunk 3]
```diff
@@ -289,20 +254,36 @@ class TOPPSiriusAdapter :
...
       MzTab csi_result;
       MzTabFile csifile;
-      CsiFingerIdMzTabWriter::read(subdirs, number, csi_result);
-      csifile.store(out2, csi_result);
+      CsiFingerIdMzTabWriter::read(subdirs, number_compounds, csi_result);
+      csifile.store(out_csifingerid, csi_result);
     }
 
-    //clean tmp directory
-    removeTempFiles_(tmp_dir, tmp_ms_file);
```
Comment (oliveralka) [inline src/utils/SiriusAdapter.cpp]: Why did you remove the void function? It is ok - just by curiosity.

Comment (lukaszimmermann) [inline src/utils/SiriusAdapter.cpp]: Probably just a matter of taste. Function is never going to called anywhere else, so I inlined the code.

**Comment by lukaszimmermann** [comment]
First error is time issue.  Second issue is short form of negation operator (which I was not aware of, will fix that). Third error I cannot access (404)

**Comment by oliveralka** [comment]
Don't worry about the negation operator - it is fine like that - even if codacy is not approving.

File: src/utils/SiriusAdapter.cpp [hunk 4]
```diff
@@ -213,17 +173,22 @@ class TOPPSiriusAdapter :
...
-    String tmp_ms_file = tmp_dir + "/" + "msfile";
-    String out_dir = tmp_dir + "/" + "sirius_out";
     SiriusMSFile::store(spectra, tmp_ms_file);
 
     //Knime hack
     QProcessEnvironment env;
     String siriuspath = "SIRIUS_PATH";
     QString qsiriuspath = env.systemEnvironment().value(siriuspath.toQString());
 
+    // TODO Is it correct that the executable argument is ignored if the SIRIUS_PATH is not empty?
```
Comment (timosachsenberg) [inline src/utils/SiriusAdapter.cpp]: @oliveralka can you comment on that?

**Comment by timosachsenberg** [comment]
looks good

**Comment by jpfeuffer** [comment]
Hmmm... I would really like to see this tested anywhere please

**Comment by timosachsenberg** [comment]
can you merge against develop again so we can see the test status ?

**Comment by lukaszimmermann** [comment]
Time out

**Pull request merged**